diff --git a/lib/include/srsran/common/common_helper.h b/lib/include/srsran/common/common_helper.h
index accd15a35..3bce2083e 100644
--- a/lib/include/srsran/common/common_helper.h
+++ b/lib/include/srsran/common/common_helper.h
@@ -60,7 +60,7 @@ inline void check_scaling_governor(const std::string& device_name)
       std::stringstream ss;
       ss << file.rdbuf();
       if (ss.str().find("performance") == std::string::npos) {
-        printf(
+        fprintf(stderr,
             "WARNING: cpu%d scaling governor is not set to performance mode. Realtime processing could be compromised. "
             "Consider setting it to performance mode before running the application.\n",
             cpu);
