diff --git a/srsue/src/stack/upper/nas.cc b/srsue/src/stack/upper/nas.cc
index f5677e631..29a12e1a5 100644
--- a/srsue/src/stack/upper/nas.cc
+++ b/srsue/src/stack/upper/nas.cc
@@ -34,6 +34,7 @@
 #include "srsran/interfaces/ue_usim_interfaces.h"
 #include "srsue/hdr/stack/upper/nas.h"
 #include "srsue/hdr/stack/upper/nas_idle_procedures.h"
+#include <signal.h>
 
 using namespace srsran;
 
@@ -211,7 +212,7 @@ void nas::timer_expired(uint32_t timeout_id)
     // Section 5.5.1.2.6 case c)
     attach_attempt_counter++;
 
-    srsran::console("Attach failed (attempt %d/%d)\n", attach_attempt_counter, max_attach_attempts);
+    //srsran::console("Attach failed (attempt %d/%d)\n", attach_attempt_counter, max_attach_attempts);
     if (attach_attempt_counter < max_attach_attempts) {
       logger.warning("Timer T3410 expired after attach attempt %d/%d: starting T3411",
                      attach_attempt_counter,
@@ -329,7 +330,6 @@ void nas::start_attach_request(srsran::establishment_cause_t cause_)
     logger.error("Couldn't allocate PDU in %s().", __FUNCTION__);
     return;
   }
-
   gen_attach_request(msg);
   if (rrc->is_connected()) {
     rrc->write_sdu(std::move(msg));
@@ -679,7 +679,7 @@ void nas::select_plmn()
     std::string debug_str = "Could not find Home PLMN Id=" + home_plmn.to_string() +
                             ", trying to connect to PLMN Id=" + known_plmns[0].to_string();
     logger.info("%s", debug_str.c_str());
-    srsran::console("%s\n", debug_str.c_str());
+    //srsran::console("%s\n", debug_str.c_str());
     current_plmn = known_plmns[0];
     state.set_deregistered(emm_state_t::deregistered_substate_t::normal_service);
   }
