diff --git a/srsue/hdr/stack/upper/nas.h b/srsue/hdr/stack/upper/nas.h
index 4e52b2309..b414f535d 100644
--- a/srsue/hdr/stack/upper/nas.h
+++ b/srsue/hdr/stack/upper/nas.h
@@ -131,7 +131,7 @@ private:
   bool         auth_request    = false;
   uint8_t      current_sec_hdr = LIBLTE_MME_SECURITY_HDR_TYPE_PLAIN_NAS;
 
-  const uint32_t max_attach_attempts    = 5; // Sec. 5.5.1.2.6
+  const uint32_t max_attach_attempts    = 1; // Sec. 5.5.1.2.6
   uint32_t       attach_attempt_counter = 0;
 
   uint32_t ip_addr       = 0;
