diff --git a/lib/src/phy/common/phy_common.c b/lib/src/phy/common/phy_common.c
index c50f2d0d6..b4c83f7c6 100644
--- a/lib/src/phy/common/phy_common.c
+++ b/lib/src/phy/common/phy_common.c
@@ -731,10 +731,10 @@ int srsran_band_get_fd_band(uint32_t         band,
     return SRSRAN_ERROR;
   }
   if (end_earfcn == -1) {
-    end_earfcn = lte_bands[i + 1].dl_earfcn_offset - 1;
+    end_earfcn = lte_bands[i + 1].dl_earfcn_offset;
   } else {
-    if (end_earfcn > lte_bands[i + 1].dl_earfcn_offset - 1) {
-      ERROR("Error: Invalid end earfcn %d. Max is %d", end_earfcn, lte_bands[i + 1].dl_earfcn_offset - 1);
+    if (end_earfcn > lte_bands[i + 1].dl_earfcn_offset) {
+      ERROR("Error: Invalid end earfcn %d. Max is %d", end_earfcn, lte_bands[i + 1].dl_earfcn_offset);
       return SRSRAN_ERROR;
     }
   }
