diff --git a/srsue/src/stack/mac/proc_ra.cc b/srsue/src/stack/mac/proc_ra.cc
index 44c6524a9..7d15fea8a 100644
--- a/srsue/src/stack/mac/proc_ra.cc
+++ b/srsue/src/stack/mac/proc_ra.cc
@@ -154,8 +154,8 @@ void ra_proc::step(uint32_t tti_)
  */
 void ra_proc::state_pdcch_setup()
 {
-  phy_interface_mac_lte::prach_info_t info = phy_h->prach_get_info();
-  if (info.is_transmitted) {
+  //phy_interface_mac_lte::prach_info_t info = phy_h->prach_get_info();
+  /*if (info.is_transmitted) {
     ra_tti  = info.tti_ra;
     ra_rnti = 1 + (ra_tti % 10) + (10 * info.f_id);
     rInfo("seq=%d, ra-rnti=0x%x, ra-tti=%d, f_id=%d", sel_preamble.load(), ra_rnti, info.tti_ra, info.f_id);
@@ -164,9 +164,9 @@ void ra_proc::state_pdcch_setup()
     rar_window_st   = ra_tti + 3;
     rntis->rar_rnti = ra_rnti;
     state           = RESPONSE_RECEPTION;
-  } else {
+  } else {*/
     rDebug("preamble not yet transmitted");
-  }
+  //}
 }
 
 /* Waits for RAR reception. rar_received variable will be set by tb_decoded_ok() function which is called when a DL
