diff --git a/lib/src/radio/radio.cc b/lib/src/radio/radio.cc
index a89e013e9..b20ef9722 100644
--- a/lib/src/radio/radio.cc
+++ b/lib/src/radio/radio.cc
@@ -476,7 +476,7 @@ bool radio::open_dev(const uint32_t& device_idx, const std::string& device_name,
     dev_name = (char*)device_name.c_str();
   }
 
-  srsran::console("Opening %d channels in RF device=%s with args=%s\n",
+  fprintf(stderr, "Opening %d channels in RF device=%s with args=%s\n",
                   nof_channels_x_dev,
                   dev_name ? dev_name : "default",
                   dev_args ? dev_args : "default");
