diff --git a/lib/src/phy/rf/rf_dev.h b/lib/src/phy/rf/rf_dev.h
index 647c7c5ba..79733bdae 100644
--- a/lib/src/phy/rf/rf_dev.h
+++ b/lib/src/phy/rf/rf_dev.h
@@ -167,7 +167,7 @@ static rf_dev_t dev_soapy = {"soapy",
                              rf_soapy_suppress_stdout,
                              rf_soapy_register_error_handler,
                              rf_soapy_open,
-                             rf_soapy_open_multi,
+                             .srsran_rf_open_multi = rf_soapy_open_multi,
                              rf_soapy_close,
                              rf_soapy_set_rx_srate,
                              rf_soapy_set_rx_gain,
@@ -243,12 +243,12 @@ static rf_dev_t dev_dummy = {"dummy",   dummy_fnc, dummy_fnc, dummy_fnc, dummy_f
 
 static rf_dev_t* available_devices[] = {
 
-#ifdef ENABLE_UHD
-    &dev_uhd,
-#endif
 #ifdef ENABLE_SOAPYSDR
     &dev_soapy,
 #endif
+#ifdef ENABLE_UHD
+    &dev_uhd,
+#endif
 #ifdef ENABLE_BLADERF
     &dev_blade,
 #endif
