diff --git a/srsue/hdr/stack/rrc/rrc.h b/srsue/hdr/stack/rrc/rrc.h
index 594967d1e..7d76b766b 100644
--- a/srsue/hdr/stack/rrc/rrc.h
+++ b/srsue/hdr/stack/rrc/rrc.h
@@ -112,6 +112,10 @@ public:
                        const T&                     msg,
                        const std::string&           msg_type);
 
+  template <class T>
+  void print_sib(int      sib_index,
+                 const T& msg);
+
   std::string print_mbms();
   bool        mbms_service_start(uint32_t serv, uint32_t port);
 
