diff --git a/lib/include/srsran/common/signal_handler.h b/lib/include/srsran/common/signal_handler.h
index 03b9a356d..b141a472c 100644
--- a/lib/include/srsran/common/signal_handler.h
+++ b/lib/include/srsran/common/signal_handler.h
@@ -57,7 +57,6 @@ static void srsran_signal_handler(int signal)
       // all other registered signals try to stop the app gracefully
       if (running) {
         running = false;
-        fprintf(stdout, "Stopping ..\n");
         alarm(SRSRAN_TERM_TIMEOUT_S);
       } else {
         // already waiting for alarm to go off ..
