diff --git a/srsue/ue.conf.example b/srsue/ue.conf.example
index 7c1cf96e9..560741054 100644
--- a/srsue/ue.conf.example
+++ b/srsue/ue.conf.example
@@ -24,8 +24,8 @@
 #####################################################################
 [rf]
 freq_offset = 0
-tx_gain = 80
-#rx_gain = 40
+tx_gain = 20
+rx_gain = 40
 
 #nof_antennas = 1
 
@@ -89,7 +89,7 @@ dl_earfcn = 3350
 # nas_filename:      File path to use for NAS packet capture
 #####################################################################
 [pcap]
-enable = none
+enable = true
 mac_filename = /tmp/ue_mac.pcap
 mac_nr_filename = /tmp/ue_mac_nr.pcap
 nas_filename = /tmp/ue_nas.pcap
