diff --git a/srsue/src/ue.cc b/srsue/src/ue.cc
index 5e76e7198..0a88c71b7 100644
--- a/srsue/src/ue.cc
+++ b/srsue/src/ue.cc
@@ -40,7 +40,7 @@ namespace srsue {
 ue::ue() : logger(srslog::fetch_basic_logger("UE", false)), sys_proc(logger)
 {
   // print build info
-  std::cout << std::endl << get_build_string() << std::endl << std::endl;
+  fprintf(stderr, "\n%s\n", get_build_string().c_str());
 }
 
 ue::~ue()
@@ -165,9 +165,9 @@ int ue::init(const all_args_t& args_)
   }
 
   if (phy) {
-    srsran::console("Waiting PHY to initialize ... ");
+    fprintf(stderr, "Waiting PHY to initialize ... ");
     phy->wait_initialize();
-    srsran::console("done!\n");
+    fprintf(stderr, "done!\n");
   }
 
   return ret;
