diff --git a/lib/examples/cell_search.c b/lib/examples/cell_search.c
index 60f78a62c..5e46ea33d 100644
--- a/lib/examples/cell_search.c
+++ b/lib/examples/cell_search.c
@@ -61,7 +61,6 @@ struct cells {
   int           dl_earfcn;
   float         power;
 };
-struct cells results[1024];
 
 float rf_gain = 70.0;
 char* rf_args = "";
@@ -149,7 +148,6 @@ int main(int argc, char** argv)
   int                           nof_freqs;
   srsran_earfcn_t               channels[MAX_EARFCN];
   uint32_t                      freq;
-  uint32_t                      n_found_cells = 0;
 
   srsran_debug_handle_crash(argc, argv);
 
@@ -239,32 +237,17 @@ int main(int argc, char** argv)
             ERROR("Error decoding MIB");
             exit(-1);
           }
-          if (ret == SRSRAN_UE_MIB_FOUND) {
-            printf("Found CELL ID %d. %d PRB, %d ports\n", cell.id, cell.nof_prb, cell.nof_ports);
-            if (cell.nof_ports > 0) {
-              results[n_found_cells].cell      = cell;
-              results[n_found_cells].freq      = channels[freq].fd;
-              results[n_found_cells].dl_earfcn = channels[freq].id;
-              results[n_found_cells].power     = found_cells[i].peak;
-              n_found_cells++;
-            }
+          if (ret == SRSRAN_UE_MIB_FOUND){
+            printf("Found CELL ID %d. %d PRB, %d ports, EARFCN %d\n", cell.id, cell.nof_prb, cell.nof_ports, channels[freq].id);
+            srsran_ue_cellsearch_free(&cs);
+            srsran_rf_close(&rf);
+            exit(0);
           }
         }
       }
     }
   }
 
-  printf("\n\nFound %d cells\n", n_found_cells);
-  for (int i = 0; i < n_found_cells; i++) {
-    printf("Found CELL %.1f MHz, EARFCN=%d, PHYID=%d, %d PRB, %d ports, PSS power=%.1f dBm\n",
-           results[i].freq,
-           results[i].dl_earfcn,
-           results[i].cell.id,
-           results[i].cell.nof_prb,
-           results[i].cell.nof_ports,
-           srsran_convert_power_to_dB(results[i].power));
-  }
-
   printf("\nBye\n");
 
   srsran_ue_cellsearch_free(&cs);
diff --git a/srsue/src/stack/rrc/rrc.cc b/srsue/src/stack/rrc/rrc.cc
index dc1495f3f..2a8c39475 100644
--- a/srsue/src/stack/rrc/rrc.cc
+++ b/srsue/src/stack/rrc/rrc.cc
@@ -50,8 +50,8 @@ using srsran::srb_to_lcid;
 
 namespace srsue {
 
-const static uint32_t NOF_REQUIRED_SIBS                = 4;
-const static uint32_t required_sibs[NOF_REQUIRED_SIBS] = {0, 1, 2, 12}; // SIB1, SIB2, SIB3 and SIB13 (eMBMS)
+const static uint32_t NOF_REQUIRED_SIBS                = 13;
+const static uint32_t required_sibs[NOF_REQUIRED_SIBS] = {0, 1, 2, 4, 5, 6, 7, 8, 9, 10, 11, 12}; // SIB1, SIB2, SIB3 and SIB13 (eMBMS)
 
 /*******************************************************************************
   Base functions
